content/BindingMethods/BindingMethods.q.md

```diff
@@ -0,0 +1,1656 @@
+# Prepared Statement with Binding Methods
+
++++
+
+### Part 1: Sample Code Analysis
+
+:::
+
+/// type=REPL, readonly=true, filename=[connection.php,student.php]
+
+```php
+// connection.php
+<?php
+    $host = 'localhost';
+    $db = 'LibraryDB';
+    $port = '5432';
+    $username = 'codestop';
+    $password = 'Admin01';
+
+    try {
+        $conn = new PDO("pgsql:host=$host;port=$port;dbname=$db;user=$username;password=$password");
+        if ($conn) {
+            echo "Successfully connected to the database.\n";
+        }
+    } catch (Exception $e) {
+        echo "Unable to establish a connection."; 
+    }
+?>
+```
+
+```php
+// student.php
+<?php
+    require_once("connection.php");
+
+    $data = array(
+        'min_birth_date' => '1999-01-01',
+        'max_birth_date' => '2000-01-01'
+    );
+    
+    $sql = "SELECT * FROM students WHERE birth_date >= :min_birth_date AND birth_date < :max_birth_date";
+
+    $pstmt = $conn->prepare($sql);
+    $pstmt->bindParam(':min_birth_date', $data['min_birth_date'], PDO::PARAM_STR);
+    $pstmt->bindParam(':max_birth_date', $data['max_birth_date'], PDO::PARAM_STR);
+    $pstmt->execute();
+    $data = $pstmt->fetchAll();
+
+    foreach ($data as $row) {
+        echo $row['student_id']."\t".$row['first_name']." ".$row['last_name']."\t".$row['birth_date']."\t".$row['gender']."\n";
```

Comment:
need more space to make it readable. Something like this
```
foreach ($data as $row) {
   echo $row['student_id'] . "\t";
   echo $row['first_name'] . "\t";
   echo $row['last_name'] . "\t";
   echo $row['birth_date'] . "\t";
   echo $row['gender'] . "\n";
}
```

Reply:
let's show them proper spacing when coding so they can see how easy it is to read the code.